main.nf

```diff
@@ -81,83 +123,360 @@ if( !(workflow.runName ==~ /[a-z]+_[a-z]+/) ){
+
+/*
+ * binning options
+ */
+params.skip_binning = false
+params.skip_busco = false
+params.min_contig_size = 1500
+
+/*
+ * assembly options
+ */
+params.skip_spades = false
+params.skip_spadeshybrid = false
+params.skip_megahit = false
+params.skip_quast = false
+
+/*
+ * taxonomy options
+ */
+params.centrifuge_db = false
+params.kraken2_db = false
+params.skip_krona = false
+params.cat_db = false
+
+/*
+ * long read preprocessing options
+ */
+params.skip_adapter_trimming = false
+params.keep_lambda = false
+params.longreads_min_length = 1000
+params.longreads_keep_percent = 90
+params.longreads_length_weight = 10
+// params.lambda_reference = "ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/viral/Escherichia_virus_Lambda/all_assembly_versions/GCA_000840245.1_ViralProj14204/GCA_000840245.1_ViralProj14204_genomic.fna.gz"
+params.lambda_reference = "$baseDir/assets/data/GCA_000840245.1_ViralProj14204_genomic.fna.gz"
+
+// Stage config files
+ch_multiqc_config = Channel.fromPath(params.multiqc_config)
+ch_output_docs = Channel.fromPath("$baseDir/docs/output.md")
+
 /*
  * Create a channel for input read files
  */
-Channel
-    .fromFilePairs( params.reads, size: params.singleEnd ? 1 : 2 )
-    .ifEmpty { exit 1, "Cannot find any reads matching: ${params.reads}\nNB: Path needs to be enclosed in quotes!\nNB: Path requires at least one * wildcard!\nIf this is single-end data, please specify --singleEnd on the command line." }
-    .into { read_files_fastqc }
+if(!params.skip_busco){
+    Channel
+        .fromPath( "${params.busco_reference}", checkIfExists: true )
+        .set { file_busco_db }
+} else {
+    file_busco_db = Channel.from()
+}
+
+if(params.centrifuge_db){
+    Channel
+        .fromPath( "${params.centrifuge_db}", checkIfExists: true )
+        .set { file_centrifuge_db }
+} else {
+    file_centrifuge_db = Channel.from()
+}
+
+if(params.kraken2_db){
+    Channel
+        .fromPath( "${params.kraken2_db}", checkIfExists: true )
+        .set { file_kraken2_db }
+} else {
+    file_kraken2_db = Channel.from()
+}
+
+if(params.cat_db){
+    Channel
+        .fromPath( "${params.cat_db}", checkIfExists: true )
+        .set { file_cat_db }
+} else {
+    file_cat_db = Channel.from()
+}
+
+if(!params.keep_phix) {
+    Channel
+        .fromPath( "${params.phix_reference}", checkIfExists: true )
+        .set { file_phix_db }
+}
+
+def returnFile(it) {
+// Return file if it exists
+    inputFile = file(it)
+    if (!file(inputFile).exists()) exit 1, "Missing file in TSV file: ${inputFile}, see --help for more information"
+    return inputFile
+}
+
+if(params.manifest){
+    manifestFile = file(params.manifest)
+    // extracts read files from TSV and distribute into channels
+    Channel
+        .from(manifestFile)
+        .ifEmpty {exit 1, log.info "Cannot find path file ${tsvFile}"}
+        .splitCsv(sep:'\t')
+        .map { row ->
+            def id = row[0]
+            def lr = returnFile( row[1] )
```

Comment:
Instead of this, I think you can just use `checkifExists`
```suggestion
            def lr = file( row[1], checkIfExists: true )
```

Then you can remove your custom function above 👍